public/app/features/dashboard/components/PanelEditor/state/actions.ts

```diff
@@ -48,6 +49,12 @@ export function panelEditorCleanUp(): ThunkResult<void> {
       sourcePanel.restoreModel(modifiedSaveModel);
 
       if (panelTypeChanged) {
+        dispatch(
+          toCollectionAction(
+            panelModelAndPluginReady({ panelId: sourcePanel.id, plugin: panel.plugin! }),
+            dashboard.uid
+          )
+        );
         dispatch(panelModelAndPluginReady({ panelId: sourcePanel.id, plugin: panel.plugin! }));
```

Comment:
I've left the original actions for each collection action for now to simplify debugging. I will remove these before merging to master.

Reply:
Resolved